_posts/2019-05-24-varying-slopes/varying_slopes.Rmd

```diff
@@ -0,0 +1,748 @@
+
+So as you might have guessed by now, _varying slopes_ (or _random slopes_, if you will) is the topic of today's post. Again, we take up an example from McElreath's book, and show how to accomplish the same thing with `tfprobability`.
+
+## Coffee, please
+
+Unlike the tadpole case, this time we work with simulated data. This is the data McElreath uses to introduce the _varying slopes_ modeling technique; he then goes on and applies it to one of the book's most featured datasets, the _pro-social_ (or indifferent, rather!) chimpanzees. For today, we stay with the simulated data for two reasons: First, the subject matter per se is non-trivial enough; and second, we want to keep careful track of what our model does, and whether its output is sufficiently close to the results McElreath obtained from _Stan_ ^[We won't overload this post by explicitly comparing results here, but we did that when writing the code.]. 
+
+So, the scenario is this. ^[Disclaimer: We have not verified whether this is an adequate model of the world, but it really doesn't matter either.] Cafés vary in how popular they are. In a popular café, when you order coffee, you're likely to _wait_. In a less popular café, you'll likely be served much faster. That's one thing.
+Second, all cafés tend to be more crowded in the mornings than in the afternoons. Thus in the morning, you'll wait longer than in the afternoon -- this goes for the popular as well as the less popular cafés.
+
+In terms of intercepts and slopes, we can picture the morning waits as intercepts, and the resultant afternoon waits as arising due to the slopes of the lines joining each morning and afternoon wait, respectively.
+
+So when we partially-pool _intercepts_, we have one "intercept prior" (itself constrained by a prior, of course), and a set of café-specific intercepts that will vary around it. When we partially-pool _slopes_, we have a "slope prior" reflecting the overall relationship between morning and afternoon waits, and a set of café-specific slopes reflecting the individual relationships. Cognitively, that means that if you have never been to the _Café Gerbeaud_ in Budapest but have been to cafés before, you might have a less-than-uninformed idea about how long you are going to wait; it also means that if you normally get your coffee in your favorite corner café in the mornings, and now you pass by there in the afternoon, you have an approximate idea how long it's going to take (namely, fewer minutes than in the mornings).
+
+So is that all? Actually, no. In our scenario, intercepts and slopes are related. If, at a less popular café, I always get my coffee before two minutes have passed, there is little room for improvement. At a highly popular café though, if it could easily take ten minutes in the mornings, then there is quite some potential for decrease in waiting time in the afternoon. So in my prediction for this afternoon's waiting time, I should factor in this interaction effect. 
```

Comment:
```suggestion
So is that all? Actually, no. In our scenario, intercepts and slopes are related. If, at a less popular café, I always get my coffee before two minutes have passed, there is little room for improvement. At a highly popular café though, if it could easily take ten minutes in the mornings, then there is potential to decrease in waiting time in the afternoon. So, in my prediction for this afternoon's waiting time, I should factor in this interaction effect. 
```